src/Bitrix24/Bitrix24Provider.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace SocialiteProviders\Bitrix24;
+
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Bitrix24Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'BITRIX24';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [''];
+
+    public static function additionalConfigKeys()
+    {
+        return ['endpoint'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase($this->getPortalUrl().'/oauth/authorize', $state);
+    }
+
+    /**
+     * Get the portal URL.
+     *
+     * @return string
+     */
+    protected function getPortalUrl()
+    {
+        return $this->getConfig('endpoint');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return 'https://oauth.bitrix.info/oauth/token/';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get($this->getPortalUrl().'/rest/user.current/', [
+            'query' => [
```

Comment:
Can you use `\GuzzleHttp\RequestOptions::QUERY` constant here instead?

Reply:
done